aws-cpp-sdk-s3-crt/source/S3CrtClient.cpp

```diff
@@ -207,6 +207,9 @@ void S3CrtClient::init(const S3Crt::ClientConfiguration& config, const std::shar
   m_crtCredProvider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderDelegate({
      std::bind([](const std::shared_ptr<AWSCredentialsProvider>& provider) {
          AWSCredentials credentials = provider->GetAWSCredentials();
```

Comment:
would you mind checking for this to be nullptr before dereferencing?

Reply:
Yeah i can do that